src/main/java/dev/the_fireplace/unforgivingvoid/mixin/ServerPlayerEntityMixin.java

```diff
@@ -36,22 +38,27 @@ private void tick(CallbackInfo callbackInfo) {
         DimensionConfig dimensionConfig = DIContainer.get().getInstance(DimensionConfigManager.class).getSettings(this.world.getRegistryKey().getValue());
         if (!this.world.isClient()
             && dimensionConfig.isEnabled()
-            && this.getBlockPos().getY() <= getBottomY(world) - dimensionConfig.getTriggerDistance()
+            && this.getBlockPos().getY() <= world.getBottomY() - dimensionConfig.getTriggerDistance()
             && !isInTeleportationState()
         ) {
             MinecraftServer server = getServer();
             if (server != null) {
                 UnforgivingVoidConstants.getLogger().debug(
                     "Player is below the minimum height. Teleporting to new dimension. Current position is {}, and current world is {}",
                     getBlockPos().toShortString(),
-                    getWorld().getRegistryKey().getValue()
+                    getServerWorld().getRegistryKey().getValue()
                 );
+
+                inTeleportationState = true;
+
                 DIContainer.get().getInstance(QueueVoidTransfer.class).queueTransfer((ServerPlayerEntity) (Object) this, server);
             }
         }
     }
 
-    private int getBottomY(World world) {
-        return world.getBottomY();
-    }
```

Comment:
I would prefer to keep this helper function, it allows changes to be more easily cherry picked back to versions that don't have `world.getBottomY()`

Reply:
Alright, I'll revert that